src/app/dev/platforms/desktop/DevToys.MauiBlazor/Shared/MainLayout.razor

```diff
@@ -1,23 +1,58 @@
-﻿@inherits MefLayoutComponentBase
+﻿@using DevToys.Core.Tools.ViewItems
 
-<FluentDesignSystemProvider 
-    AccentBaseColor="#0078D4"
-    NeutralBaseColor="000000"
-    BaseLayerLuminance=@_baseLayerLuminanceValue>
-    <div id="container">
-        <NavigationView>
-            <MenuItemsSource>
-                <FluentButton Appearance="Appearance.Stealth">Hello</FluentButton>
-                <FluentButton Appearance="Appearance.Stealth">There</FluentButton>
-            </MenuItemsSource>
-            <FooterMenuItemsSource>
-                <FluentButton Appearance="Appearance.Stealth">Bottom</FluentButton>
-            </FooterMenuItemsSource>
-            <Content>
-                <main>
-                    @Body
-                </main>
-            </Content>
-        </NavigationView>
-    </div>
-</FluentDesignSystemProvider>
\ No newline at end of file
+@using Microsoft.Fast.Components.FluentUI
+@using Microsoft.Fast.Components.FluentUI.DesignTokens
+
+@inherits MefLayoutComponentBase
+
+<header>
+</header>
+
+<Navbar Id="navbar" IsCollapsible="true" HasSearch="false" IsVertical="true">
+    <NavbarItems IsScrollable="true" IsScrollableVertical="true">
+        @foreach (var navItem in ViewModel.HeaderAndBodyToolViewItems)
+        {
+            if (navItem is GuiToolViewItem guiToolViewItem)
+            {
+                <NavbarItem Page="@guiToolViewItem.InternalName" Icon="@(FluentIcons.Home)" Text="@guiToolViewItem.MenuDisplayTitle" />
+            }
+            else if (navItem is GroupViewItem groupViewItem)
+            {
+                if (groupViewItem.Children is not null)
+                {
+                    <NavbarGroup Page="@groupViewItem.InternalName" Icon="@groupViewItem.IconGlyph" Text="@groupViewItem.DisplayTitle">
+                        @foreach (var childItem in groupViewItem.Children)
+                        {
+                            <NavbarItem Page="@childItem.InternalName" Icon="@(FluentIcons.Home)" Text="@childItem.MenuDisplayTitle" />
+                        }
+                    </NavbarGroup>
+                }
+                else
+                {
+                    <NavbarItem Page="@groupViewItem.InternalName" Icon="@(FluentIcons.Home)" Text="@groupViewItem.DisplayTitle" />
+                }
+            }
+            else
+            {
+                <li class="nav-item nav-separator"></li>
+            }
+        }
+    </NavbarItems>
+    <NavbarItems IsFixedBottom="true">
+        @foreach (var navItem in ViewModel.FooterToolViewItems)
+        {
+            if (navItem is GuiToolViewItem guiToolViewItem)
+            {
+                <NavbarItem Page="@guiToolViewItem.InternalName" Icon="@(FluentIcons.Home)" Text="@guiToolViewItem.MenuDisplayTitle" />
+            }
+            else
+            {
+                <li class="nav-item nav-separator"></li>
+            }
+        }
+    </NavbarItems>
+</Navbar>
+
+<main class="page">
+    @Body
```

Comment:
I'm playing a bit with the code to try to make the components for the tools. I notice that for now we only have `Index` and it uses the URI to determine the tool name to display.

Thinking about it, I thought `OK but we don't have only tools, sometimes we also display a list of tools in this area, which isn't a tool itself`. So naive as I am, I started making a page `ToolPage.razor` and `ToolGroupPage.razor`, exactly like in the WASDK app. One page would have the URI `tool/{Name}` and the other `group/{Name}` but then I quickly realized that you did it this way because when clicking on a menu item, the whole page would refresh when navigating to a different URL, which is a problem indeed....

I'm not fan of the idea of mixing the Tool UI and the Group UI in the same page (`Index.razor`).

So how about the following:
Instead of navigating to a page, we listen to a `MainWindowViewModel.SelectedMenuItem` and it change, depending on whether it's a `GuiToolViewItem` or `GroupViewItem`, we show here a `ToolPage` component or a `ToolGroupPage` component. This way we never navigate anywhere and just replace the body on the fly. In addition, this way we pass the whole `GuiToolViewItem` to the `ToolPage` instead of relying on the internal name to find which tool / group we're trying to display.

Would this work? Would this have any performance impact compared to navigating (which, if I understood well, isn't really navigating at the moment since we don't refresh the whole page, right?) ?
What do you think? :)

Reply:
PS: This could be done in a different PR. Happy to do it myself too if you want, since I kind of need this first before creating the tool's UI component.